core/tests/performance_framework/process_data.py

```diff
@@ -0,0 +1,147 @@
+
+    page_session_stats is a dictionary containing page load statistics from an
+    HTTP Archive.
+    (https://dvcs.w3.org/hg/webperf/raw-file/tip/specs/HAR/Overview.html)
+
+    page_session_timings is a dictionary containing metrics associated with page
+    loading and includes the following keys:
+
+        timing: maps to a dict containing the keys (all in milliseconds):
+            domLoading: the Unix time the DOM started loading
+            fetchStart: the Unix time the page began loading
+            responseStart: the Unix time the response started
+            loadEventEnd: the Unix time the page finished loading
+            requestStart: the Unix time the request started
+            responseEnd: the Unix time the request finished
+            domComplete: the Unix time the DOM finished loading
+            domInteractive: the Unix time the the parser finished its work on
+                            the main document
+    """
+
+    TIMING_PROPERTIES = [
+        'loadEventEnd', 'fetchStart', 'domComplete', 'domInteractive',
+        'responseEnd', 'requestStart'
+        ]
+
+    def __init__(self, page_session_stats=None, page_session_timings=None):
+        self.page_session_stats = page_session_stats
+        self.page_session_timings = page_session_timings
+
+        self.page_load_timings = {}
+        if self.page_session_timings:
+            self.page_load_timings = page_session_timings['timing']
+
+        self._validate()
+
+    def _validate(self):
+        """Validates various properties of a PageSessionMetrics object."""
+        if not self.page_session_stats and not self.page_session_timings:
+            raise utils.ValidationError(
+                'Expected atleast one argument among page_session_stats or '
+                'page_session_timings to be present.')
+
+        if self.page_session_stats:
+            if 'log' not in self.page_session_stats:
+                raise utils.ValidationError(
+                    'Expected the page load stats to have a \'log\' entry')
+
+            if 'entries' not in self.page_session_stats['log']:
+                raise utils.ValidationError(
+                    'Expected the log entry of the page load stats to include'
+                    'an additional \'entries\' element')
+
+        if self.page_session_timings:
+            for timing_prop in self.TIMING_PROPERTIES:
+                if timing_prop not in self.page_load_timings:
+                    raise utils.ValidationError(
+                        'Expected the timing entry of the page load timings to'
+                        'include %s property' % timing_prop)
+
+    def get_request_count(self):
+        """Returns the number of requests made prior to the page load
+        completing."""
+        request_count = 0
+
+        request_count = len(self.page_session_stats['log']['entries'])
+
+        if request_count == 0:
+            raise Exception('Total requests cannot be 0.')
```

Comment:
Shouldn't this be in ``_validate``? Could this method be overall simplified:

``return len(self.page_session_stats['log']['entries'])``